packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -436,6 +436,8 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
   @override
   Widget build(BuildContext context) {
     assert(debugCheckHasDirectionality(context));
+
+    final double bottomPadding = math.max(MediaQuery.of(context).padding.bottom - _kBottomMargin, 0.0);
```

Comment:
FWIW: not entirely thrilled with this. An alternative would be to deal with this in the individual item widgets where we deal with bottom margin. Ideas welcome :)

Reply:
I think consolidating here is fine. Though some comment / const renaming might help future readers. Something like we only need to shift up if the partial obstruction may intersect with interactive parts of this widget, liberally generalized by this const (feel free to do all this in a different PR).

Reply:
Maybe the 2 names needs some additional differentiation too. They're both called bottom padding which is a bit confusing when one is a outer container padding and one is an inner layout padding.

Reply:
Good point. I'll update the names.

Reply:
Done.